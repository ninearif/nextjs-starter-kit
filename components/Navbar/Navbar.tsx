import React, { useEffect, useState } from 'react';
import Headroom from 'react-headroom';
import { FiGithub } from 'react-icons/fi';
type Props = {};

function Navbar({}: Props) {
  // const [offset, setOffset] = useState(0);
  // const [scrollUp, setScrollUp] = useState(false);

  // useEffect(() => {
  //   const onScroll = () => setOffset(window.pageYOffset);

  //   // remove listener if exist.
  //   window.removeEventListener('scroll', onScroll);
  //   // set an scroll event listener
  //   window.addEventListener('scroll', onScroll, { passive: true });

  //   // clean up event listener
  //   return () => window.removeEventListener('scroll', onScroll);
  // }, []);

  // useEffect(() => {
  //   console.log(offset);
  // }, [offset]);

  return (
    <Headroom>
      <nav className="w-full bg-white text-gray-700">
        <div className="navbar mx-auto md:container">
          <div className="navbar-start">
            <div className="dropdown">
              <label tabIndex={0} className="btn btn-ghost lg:hidden">
                <svg
                  xmlns="http://www.w3.org/2000/svg"
                  className="h-5 w-5"
                  fill="none"
                  viewBox="0 0 24 24"
                  stroke="currentColor"
                >
                  <path
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="2"
                    d="M4 6h16M4 12h8m-8 6h16"
                  />
                </svg>
              </label>
              <ul
                tabIndex={0}
                className="dropdown-content menu rounded-box menu-compact mt-3 w-52 bg-base-100 p-2 shadow"
              >
                <li>
                  <a>Item 1</a>
                </li>
                <li tabIndex={0}>
                  <a className="justify-between">
                    Parent
                    <svg
                      className="fill-current"
                      xmlns="http://www.w3.org/2000/svg"
                      width="24"
                      height="24"
                      viewBox="0 0 24 24"
                    >
                      <path d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z" />
                    </svg>
                  </a>
                  {/* <ul className="p-2">
                    {navbarItems.map((item) => {
                      return (
                        <li key={'navbar-' + item.href}>
                          <a href={item.href}>{item.text}</a>
                        </li>
                      );
                    })}
                  </ul> */}
                </li>
                <li>
                  <a>Item 3</a>
                </li>
              </ul>
            </div>
            <a className="btn btn-ghost text-2xl normal-case">Arif.</a>
          </div>
          <div className="navbar-center hidden lg:flex">
            {/* <ul className="menu menu-horizontal p-0">
              {navbarItems.map((item) => {
                return (
                  <li key={'navbar-' + item.href}>
                    <a href={item.href}>{item.text}</a>
                  </li>
                );
              })} }
            </ul>*/}
          </div>
          <div className="navbar-end">
            <a className="btn btn-primary">
              <FiGithub className="md:mr-2" />
              <span className="hidden md:block">View on Github</span>
            </a>
          </div>
        </div>
      </nav>
    </Headroom>
  );
}

export default Navbar;
