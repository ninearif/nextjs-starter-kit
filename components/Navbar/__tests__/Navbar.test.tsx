import { render, screen } from '@testing-library/react';

import Navbar from '../Navbar';

describe('Navbar component test', () => {
  it('shoud include github button', () => {
    render(<Navbar />);
    expect(screen.findByRole('button')).resolves.toHaveTextContent(
      'VIEW ON GITHUB'
    );
  });
});

describe('Navbar component snapshot test', () => {
  it('render Navbar unchanged', () => {
    const { container } = render(<Navbar />);
    expect(container).toMatchSnapshot();
  });
});
