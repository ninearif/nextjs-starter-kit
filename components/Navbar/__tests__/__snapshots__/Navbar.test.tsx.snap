// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component snapshot test render Navbar unchanged 1`] = `
<div>
  <div
    class="headroom-wrapper"
  >
    <div
      class="headroom headroom--unfixed"
      style="position: relative; top: 0px; left: 0px; right: 0px; z-index: 1; transform: translate3D(0, 0, 0);"
    >
      <nav
        class="w-full bg-white text-gray-700"
      >
        <div
          class="navbar mx-auto md:container"
        >
          <div
            class="navbar-start"
          >
            <div
              class="dropdown"
            >
              <label
                class="btn btn-ghost lg:hidden"
                tabindex="0"
              >
                <svg
                  class="h-5 w-5"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4 6h16M4 12h8m-8 6h16"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </label>
              <ul
                class="dropdown-content menu rounded-box menu-compact mt-3 w-52 bg-base-100 p-2 shadow"
                tabindex="0"
              >
                <li>
                  <a>
                    Item 1
                  </a>
                </li>
                <li
                  tabindex="0"
                >
                  <a
                    class="justify-between"
                  >
                    Parent
                    <svg
                      class="fill-current"
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
                      />
                    </svg>
                  </a>
                </li>
                <li>
                  <a>
                    Item 3
                  </a>
                </li>
              </ul>
            </div>
            <a
              class="btn btn-ghost text-2xl normal-case"
            >
              Arif.
            </a>
          </div>
          <div
            class="navbar-center hidden lg:flex"
          />
          <div
            class="navbar-end"
          >
            <a
              class="btn btn-primary"
            >
              <svg
                class="md:mr-2"
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"
                />
              </svg>
              <span
                class="hidden md:block"
              >
                View on Github
              </span>
            </a>
          </div>
        </div>
      </nav>
    </div>
  </div>
</div>
`;
